ecommerce/baskets/views.py

```diff
@@ -15,12 +20,45 @@ class BasketItemViewSet(DetailedViewSetMixin, viewsets.ModelViewSet):
         "detailed": BasketItemDetailedSerializer,
     }
 
+    def get_queryset(self):
+        queryset = super().get_queryset()
+        user_id = self.request.user.id
+        # import pdb; pdb.set_trace()
+        return queryset.filter(basket__customer__id=user_id)
+
 
 class BasketViewSet(DetailedViewSetMixin, viewsets.ModelViewSet):
-    queryset = Basket.objects.all()
-    serializer_class = BasketSerializer
-    filterset_class = BasketFilter
+    http_method_names = ["get", "delete"]
+    permission_classes = ()
+    queryset = BasketItem.objects.all()
+    serializer_class = BasketItemDetailedSerializer
+    filterset_class = BasketItemFilter
     serializer_action_classes = {
-        "detailed_list": BasketDetailedSerializer,
-        "detailed": BasketDetailedSerializer,
+        "add_to_cart": BasketItemSerializer,
     }
+
+    def get_queryset(self):
+        queryset = super().get_queryset()
+        user_id = self.request.user.id
+        return queryset.filter(basket__customer__id=user_id)
+
+    @action(detail=False, methods=["post"], http_method_names=["post"])
+    def add_to_cart(self, request, pk=None):
+        user_id = self.request.user.id
```

Comment:
kayıtlı olmayan kullanıcı gelirse burda hata alırız.

Reply:
Evet. Diğerini başta yapamadım. İlk sürüm olarak giriş yapılıyken de olsa yapılabilmesi için böyle yaptım. Gün içinde düzeltmeye çalışacağım.

Reply:
> kayıtlı olmayan kullanıcı gelirse burda hata alırız.

Evet. Diğerini başta yapamadım. İlk sürüm olarak giriş yapılıyken de olsa yapılabilmesi için böyle yaptım. Gün içinde düzeltmeye çalışacağım.